### ISSUE TITLE
Automatically generate warp_to entries in data/maps/objects somehow

### ISSUE THREAD

**Issue by Rangi42** [pret/pokered#286]
Every `warp` has a corresponding `warp_to`, using the coordinates of the warp and the width of the current map. Ideally there could be a `finish_warps` or `finish_warps WIDTH` macro that would generate them all without having to manually keep them in sync. This may not be possible with current rgbasm features (i.e. no arrays).

**Comment by Rangi42** [comment]
This might be possible with (ab)use of string functions like [rgbds-structs](https://github.com/ISSOtm/rgbds-structs), if that doesn't end up being too complex/messy.

**Comment by Rangi42** [comment]
If a `def_warps` macro existed instead of the `db # ; warps` count, like pokecrystal does, then this would work:

```diff
 def_warps: MACRO
 	IF DEF(_NUM_WARPS)
 		PURGE _NUM_WARPS
 	ENDC
 _NUM_WARPS EQUS "_NUM_WARPS_\@"
 	db _NUM_WARPS
 _NUM_WARPS = 0
+	IF DEF(_WARP_TO_BUFFER)
+		PURGE _WARP_TO_BUFFER
+	ENDC
+_WARP_TO_BUFFER EQUS ""
 ENDM
 
 warp: MACRO
 	db \2, \1, \3, \4
 _NUM_WARPS = _NUM_WARPS + 1
+_TMP EQUS "{_WARP_TO_BUFFER}\n\tevent_displacement _FINISH_WARPS_WIDTH, \1, \2"
+	PURGE _WARP_TO_BUFFER
+_WARP_TO_BUFFER EQUS "{_TMP}"
+	PURGE _TMP
 ENDM
+
+finish_warps: MACRO
+_FINISH_WARPS_WIDTH EQU \1
+	_WARP_TO_BUFFER
+PURGE _FINISH_WARPS_WIDTH
+ENDM
```

Then every block of `warp_to`s could be replaced by `finish_warps THE_MAP_WIDTH`.

**Comment by Rangi42** [comment]
Resolved by #287.
